dom.go

```diff
@@ -1946,120 +1992,196 @@ func (e *HTMLCanvasElement) GetContext(param string) *js.Object {
 	return e.Call("getContext", param)
 }
 
-// Colors, Styles, and Shadows
+// Drawing Rectangles
 
-func (ctx *CanvasRenderingContext2D) CreateLinearGradient(x0, y0, x1, y1 int) {
-	ctx.Call("createLinearGradient", x0, y0, x1, y1)
+func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height float64) {
+	ctx.Call("clearRect", x, y, width, height)
 }
 
-// Rectangles
+func (ctx *CanvasRenderingContext2D) FillRect(x, y, width, height float64) {
+	ctx.Call("fillRect", x, y, width, height)
+}
 
-func (ctx *CanvasRenderingContext2D) Rect(x, y, width, height int) {
-	ctx.Call("rect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height float64) {
+	ctx.Call("strokeRect", x, y, width, height)
 }
 
-func (ctx *CanvasRenderingContext2D) FillRect(x, y, width, height int) {
-	ctx.Call("fillRect", x, y, width, height)
+// Drawing Text
+
+func (ctx *CanvasRenderingContext2D) FillText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("fillText", text, x, y)
+		return
+	}
+
+	ctx.Call("fillText", text, x, y, maxWidth)
 }
 
-func (ctx *CanvasRenderingContext2D) StrokeRect(x, y, width, height int) {
-	ctx.Call("strokeRect", x, y, width, height)
+func (ctx *CanvasRenderingContext2D) StrokeText(text string, x, y, maxWidth float64) {
+	if maxWidth == -1 {
+		ctx.Call("strokeText", text, x, y)
+		return
+	}
+
+	ctx.Call("strokeText", text, x, y, maxWidth)
+}
+func (ctx *CanvasRenderingContext2D) MeasureText(text string) *TextMetrics {
+	textMetrics := ctx.Call("measureText", text)
+	return &TextMetrics{Object: textMetrics}
 }
 
-func (ctx *CanvasRenderingContext2D) ClearRect(x, y, width, height int) {
-	ctx.Call("clearRect", x, y, width, height)
+// Line styles
+
+func (ctx *CanvasRenderingContext2D) GetLineDash() interface{} {
+	return ctx.Call("getLineDash")
 }
 
-// Paths
+//
+func (ctx *CanvasRenderingContext2D) SetLineDash(arr interface{}) {
```

Comment:
According to https://godoc.org/github.com/gopherjs/gopherjs/js:

```
| Go type               | JavaScript type       | Conversions back to interface{} |
| integers and floats   | Number                | float64
| --------------------- | --------------------- | ------------------------------- |
| []float64             | Float64Array          | []float64                       |
| all other slices      | Array                 | []interface{}                   |
```

So `[]int` would get encoded as `Array` of `Integer`s.

Do we need to allow both `[]int` and `[]float64` as inputs? Then no choice but to use `interface{}`, but we should document the expected types.

Are you sure `float64` inputs are valid? The examples only use integer values. Is it possible to have a segment with a floating point value? Just want to confirm, so we can create the best API possible here.